port/win/port_win.h

```diff
@@ -81,8 +81,8 @@ namespace rocksdb {
 
 namespace port {
 
-// VS 15
-#if (defined _MSC_VER) && (_MSC_VER >= 1900)
+// VS 15 / MinGW
+#if !defined(_MSC_VER) || (_MSC_VER >= 1900)
```

Comment:
@orgads looks like CI hasn't started on this yet (and won't for a while). Perhaps it's worth rewriting it in the style I suggested above, for clarity?